CUDA library wrapper fails to compile on certain clang versions

~~Clang 21~~ Clang 22 fails to build these for me, can be trivially reproduced with the `Data/nix/cmake_enable_libfwd.sh` script:
```
libcuda_Guest.cpp:59:48: error: constexpr variable 'proc_override' must be initialized by a constant expression
   59 | constexpr static std::array<override_entry, 2> proc_override = {
      |                                                ^               ~
   60 |   {{"cuGetProcAddress", (void*)cuGetProcAddress_v2}, {"cuGetProcAddress_v2", (void*)cuGetProcAddress_v2}}};
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libcuda_Guest.cpp:60:25: note: cast that performs the conversions of a reinterpret_cast is not allowed in a constant expression
   60 |   {{"cuGetProcAddress", (void*)cuGetProcAddress_v2}, {"cuGetProcAddress_v2", (void*)cuGetProcAddress_v2}}};
```

Originally mentioned in https://github.com/FEX-Emu/FEX/pull/5536#discussion_r3429122873

--- a/ThunkLibs/libcuda/libcuda_Guest.cpp
+++ b/ThunkLibs/libcuda/libcuda_Guest.cpp
@@ -56,7 +56,7 @@ struct override_entry {
   void* ptr;
 };
 
-constexpr static std::array<override_entry, 2> proc_override = {
+const static std::array<override_entry, 2> proc_override = {
   {{"cuGetProcAddress", (void*)cuGetProcAddress_v2}, {"cuGetProcAddress_v2", (void*)cuGetProcAddress_v2}}};
 
 CUresult cuGetProcAddress_v2(const char* symbol, void** pfn, int cudaVersion, cuuint64_t flags, CUdriverProcAddressQueryResult* symbolStatus) {

--- a/ThunkLibs/libcuda/libcuda_Host.cpp
+++ b/ThunkLibs/libcuda/libcuda_Host.cpp
@@ -17,7 +17,7 @@ struct override_entry {
   void* ptr;
 };
 
-constexpr static std::array<override_entry, 2> proc_override = {
+const static std::array<override_entry, 2> proc_override = {
   {{"cuCtxCreate_v2", (void*)FEXFN_IMPL(cuCtxCreate_v2)}, {"cuGetExportTable", (void*)FEXFN_IMPL(cuGetExportTable)}}};
 
 static CUresult FEXFN_IMPL(cuGetProcAddress_v2)(const char* symbol, guest_layout<void**> pfn, int cudaVersion, cuuint64_t flags,

